server/db/sql.go

```diff
@@ -50,54 +50,61 @@ func newDBClient() *gorm.DB {
 		panic(fmt.Sprintf("Unknown DB Dialect: '%s'", dbConfig.Dialect))
 	}
 
-	err := dbClient.AutoMigrate(
-		&models.Beacon{},
-		&models.BeaconTask{},
-		&models.DNSCanary{},
-		&models.Crackstation{},
-		&models.Benchmark{},
-		&models.CrackTask{},
-		&models.CrackCommand{},
-		&models.CrackFile{},
-		&models.CrackFileChunk{},
-		&models.Certificate{},
-		&models.Host{},
-		&models.HttpC2Config{},
-		&models.HttpC2ImplantConfig{},
-		&models.HttpC2ServerConfig{},
-		&models.HttpC2Header{},
-		&models.HttpC2PathSegment{},
-		&models.HttpC2Cookie{},
-		&models.HttpC2URLParameter{},
-		&models.IOC{},
-		&models.ExtensionData{},
-		&models.ImplantBuild{},
-		&models.ImplantProfile{},
-		&models.ImplantConfig{},
-		&models.ImplantC2{},
-		&models.EncoderAsset{},
-		&models.KeyExHistory{},
-		&models.KeyValue{},
-		&models.CanaryDomain{},
-		&models.Loot{},
-		&models.Credential{},
-		&models.Operator{},
-		&models.Website{},
-		&models.WebContent{},
-		&models.WGKeys{},
-		&models.WGPeer{},
-		&models.ListenerJob{},
-		&models.HTTPListener{},
-		&models.DNSListener{},
-		&models.WGListener{},
-		&models.MultiplayerListener{},
-		&models.MtlsListener{},
-		&models.DnsDomain{},
-		&models.MonitoringProvider{},
-		&models.ResourceID{},
-	)
-	if err != nil {
-		clientLog.Error(err)
+	// We cannot pass all of these into AutoMigrate at once because if one fails, subsequent models will not be created
+	var allDBModels []interface{} = make([]interface{}, 0)
+
+	allDBModels = append(allDBModels, &models.HttpC2Header{})
```

Comment:
Instead of calling `append` on every line, it's probably best to have a static declaration of `allDBModels`:

```go

var allDBModels []interface{} = {
&models.HttpC2Header{},
....
}
```

Calling `append` on every line will reallocate a new slice every time.

Reply:
Ooo - good call - thank you!